utils/src/main/java/com/redhat/lightblue/migrator/utils/DAOFacadeBase.java

```diff
@@ -31,12 +33,40 @@
 
     protected final D legacyDAO, lightblueDAO;
 
+    private EntityIdStore entityIdStore = null;
+
+    public EntityIdStore getEntityIdStore() {
+        return entityIdStore;
+    }
+
+    public void setEntityIdStore(EntityIdStore entityIdStore) {
+        this.entityIdStore = entityIdStore;
+
+        try {
+            Method method = lightblueDAO.getClass().getMethod("setEntityIdStore", EntityIdStore.class);
+            method.invoke(lightblueDAO, entityIdStore);
```

Comment:
What about providing an interface that entities need to implement? If you are requiring the method exists here anyway, it might be cleaner.